x-pack/plugins/observability/public/components/alerts_table/slo/render_cell_value.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiLink } from '@elastic/eui';
+import React from 'react';
+import {
+  ALERT_DURATION,
+  ALERT_RULE_NAME,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_STATUS_RECOVERED,
+  ALERT_REASON,
+  TIMESTAMP,
+} from '@kbn/rule-data-utils';
+import { isEmpty } from 'lodash';
+import type {
+  DeprecatedCellValueElementProps,
+  TimelineNonEcsData,
+} from '@kbn/timelines-plugin/common';
+
+import { asDuration } from '../../../../common/utils/formatters';
+import { AlertStatusIndicator } from '../../alert_status_indicator';
+import { TimestampTooltip } from '../timestamp_tooltip';
+import { parseAlert } from '../../../pages/alerts/helpers/parse_alert';
+import type { ObservabilityRuleTypeRegistry } from '../../../rules/create_observability_rule_type_registry';
+import type { TopAlert } from '../../../typings/alerts';
+
+export const getMappedNonEcsValue = ({
+  data,
+  fieldName,
+}: {
+  data: TimelineNonEcsData[];
+  fieldName: string;
+}): string[] | undefined => {
+  const item = data.find((d) => d.field === fieldName);
+  if (item != null && item.value != null) {
+    return item.value;
+  }
+  return undefined;
+};
+
+const getRenderValue = (mappedNonEcsValue: any) => {
+  // can be updated when working on https://github.com/elastic/kibana/issues/140819
+  const value = Array.isArray(mappedNonEcsValue) ? mappedNonEcsValue.join() : mappedNonEcsValue;
+
+  if (!isEmpty(value)) {
+    if (typeof value === 'object') {
+      return JSON.stringify(value);
+    }
+    return value;
+  }
+
+  return '—';
```

Comment:
nit -> normally we do that
```suggestion
  return '--';
```